### ISSUE TITLE
Lambda PrincipalWithConditions error when using CDK 2.30.0

### ISSUE THREAD

**Issue by hutchy2570** [alma-cdk/openapix#23]
Hi, 

Firstly thanks for building this library, having an openapi driven API Gateway has always been a bit rough when using Cloudformation/SAM/CDK and openapix seems to be a great idea to make this a much nicer situation!

I've been evaluating using openapix and have come across an issue with Lambda backed apis when using CDK 2.30.0 (released 3 days ago as of writing).  When I use the `hello-api` example and upgrade the `aws-cdk` and `aws-cdk-lib` dependencies to `2.30.0` I receive the following error during synth.

```
PrincipalWithConditions had unsupported conditions for Lambda permission statement: [{"operator":"0","key":"ArnLike"}]. Supported operator/condition pairs: [{"operator":"ArnLike","key":"aws:SourceArn"},{"operator":"StringEquals","key":"aws:SourceAccount"},{"operator":"StringEquals","key":"aws:PrincipalOrgID"}]
```

It works fine with `2.29.1`.

I can see in the [changelog for CDK](https://github.com/aws/aws-cdk/releases/tag/v2.30.0) that there's been some changes around Lambda and API Gateway permissions, so I can only assume these have caused an incompatibility with openapix.

Edit:

I somehow missed this issue when looking in the CDK repo, looks like a regression on CDK's side. https://github.com/aws/aws-cdk/issues/20974

**Comment by hutchy2570** [comment]
As a workaround, you can use `patch-package` with the following diff to fix the current version (0.36.0 at the time of writing).  I've verified this works with CDK `2.32.1`.

`patches/@alma-cdk+openapix+0.0.36.patch`
```
diff --git a/node_modules/@alma-cdk/openapix/lib/api/api.js b/node_modules/@alma-cdk/openapix/lib/api/api.js
index 7a420ff..2b0d238 100644
--- a/node_modules/@alma-cdk/openapix/lib/api/api.js
+++ b/node_modules/@alma-cdk/openapix/lib/api/api.js
@@ -71,13 +71,11 @@ class Api extends aws_apigateway_1.SpecRestApi {
     /** Resolve ARN Service Principal for given API Gateway instance. */
     getApiGatewayPrincipal(api) {
         return new aws_iam_1.ServicePrincipal('apigateway.amazonaws.com', {
-            conditions: [
-                {
-                    ArnLike: {
-                        'AWS:SourceArn': `${api.arnForExecuteApi}/*`,
-                    },
+            conditions: {
+                ArnLike: {
+                    'aws:SourceArn': api.arnForExecuteApi(),
                 },
-            ],
+            },
         });
     }
     /** Determine if the integration internal type is `LAMBDA`. */
```

**Comment by aripalo** [comment]
Hi! Thanks for reporting this and sorry for the response delay (July / early August is a common summer vacation time here in Finland so we're mostly AFK during that time 😅).

We've also able to reproduce this issue with changing the CDK version used by this package to `2.30.0` which leads to an error during tests:
![Screenshot 2022-08-19 at 15 29 44](https://user-images.githubusercontent.com/679146/185618569-718fac89-3e96-4df3-9e9b-d7a12ad2fa50.png)


As you pointed out, it seems to be related to a regression on CDK side. 

Upgrading to `2.32.1` (or newer) & modifying the service principal conditions as you suggested seems to resolve the situation.

I'll create a PR next.
